Bowling.java

```diff
@@ -74,8 +74,13 @@ public String printFrame() {
 
 class Scoreboard {
 
+	private static final int strikePoints = 10;
+	private static final int sparePoints = 10;
 	private List<Frame> allFrames;
 	private Integer totalPoints;
+	private Optional<Frame> secondToLastFrame;
+	private Optional<Frame> thirdToLastFrame;
+	private Frame currentFrame;
```

Comment:
I see how that side effect would make you nervous. I think. Maybe I should echo what you said to make sure. You are nervous because its not obvious that these values will mutate. yes?

I changed it so that these variables are now being defined in `update`, but I am not sure that this provides enough clarity. Should the methods defining the variables be renamed `updateCurrentFrame` and `updateFrameBefore`? (Not sure this makes sense since there is no initial value.)

I see several other options: 
- Define them within each method, accept the duplication, and remove the member fields. You mentioned that we only have a few right now where they are being used, but I'm thinking we will have more down the line (ex. `isDouble`).
- Leave them in `updateTotalPoints`, pass them as arguments to `isTurkey` and `isSingleStrike`, and remove the member fields.

I'm not sure which is best. I aim for writing more concisely, but I know that isn't always best for clarity.